contracts/votingMachines/GenesisProtocolLogic.sol

```diff
@@ -0,0 +1,652 @@
+        }
+    }
+
+  /**
+   * @dev Check that the proposal is votable (open and not executed yet)
+   */
+    modifier votable(bytes32 _proposalId) {
+        require(_isVotable(_proposalId));
+        _;
+    }
+
+    /**
+     * @dev register a new proposal with the given parameters. Every proposal has a unique ID which is being
+     * generated by calculating keccak256 of a incremented counter.
+     * @param _paramsHash parameters hash
+     * @param _proposer address
+     * @param _organization address
+     */
+    function propose(uint , bytes32 _paramsHash,address _proposer,address _organization)
+        external
+        returns(bytes32)
+    {
+        //Check parameters existence.
+        require(parameters[_paramsHash].quedVoteRequiredPercentage > 0);
+            // Generate a unique ID:
+        bytes32 proposalId = keccak256(abi.encodePacked(this, proposalsCnt));
+        proposalsCnt++;
+         // Open proposal:
+        Proposal memory proposal;
+        proposal.callbacks = msg.sender;
+        proposal.organizationId = keccak256(abi.encodePacked(msg.sender,_organization));
+
+        proposal.state = ProposalState.Qued;
+        // solium-disable-next-line security/no-block-members
+        proposal.times[0] = now;
+        proposal.currentBoostedVotePeriodLimit = parameters[_paramsHash].boostedVotePeriodLimit;
+        proposal.proposer = _proposer;
+        proposal.winningVote = NO;
+        proposal.paramsHash = _paramsHash;
+        if (organizations[proposal.organizationId] == 0) {
+            if (_organization == address(0)) {
+                organizations[proposal.organizationId] = msg.sender;
+            } else {
+                organizations[proposal.organizationId] = _organization;
+            }
+        }
+        //calc dao bounty
+        uint daoBounty = parameters[_paramsHash].daoBountyConst.mul(averagesBoostDownstakes[proposal.organizationId]);
+        if (daoBounty < parameters[_paramsHash].minimumDaoBounty) {
+            proposal.daoBountyRemain = parameters[_paramsHash].minimumDaoBounty;
+        } else {
+            proposal.daoBountyRemain = daoBounty;
+        }
+        proposal.totalStakes = proposal.daoBountyRemain;
+        proposals[proposalId] = proposal;
+        proposals[proposalId].stakes[NO] = proposal.daoBountyRemain;//dao downstake on the proposal
+        emit NewProposal(proposalId, organizations[proposal.organizationId], NUM_OF_CHOICES, _proposer, _paramsHash);
+        return proposalId;
+    }
+
+    /**
+      * @dev expired try to execute a boosted proposal if it is expired
+      * @param _proposalId the id of the proposal
+      * @return uint expirationCallBounty the bounty amount for the expiration call
+     */
+    function expired(bytes32 _proposalId) external returns(uint expirationCallBounty) {
+        Proposal storage proposal = proposals[_proposalId];
+        require(proposal.state == ProposalState.Boosted);
+        require(_execute(_proposalId));
```

Comment:
Yes. because there is only need to count the number of the boosted proposals ...there is no need to incentivise expiration call on other proposal states.
The doc need to be updated.

Reply:
you should check the proposal expiration before call..

Reply:
that is unfortunate for the caller.  why does the `require` need to be there at all?

Reply:
In any case, if this function executes a boosted proposal, whether it expired or not, perhaps rename it (again) to "executeBoosted" instead of "executeExpriedBoosted"

Reply:
it can execute only  expired proposal.

Reply:
@orenyodfat Can we ensure that all the `require` statements in this PR have a reason string?

Reply:
* You should still rename this function, given that it does an action it should not be named as an adjective
* I am very much for adding error strings to the require statements. It is a small cost, but it will make life for devs SOOO much easier.

Reply:
see 
9fdd605